src/main/java/de/hpi/bpt/argos/api/eventType/EventTypeEndpointImpl.java

```diff
@@ -0,0 +1,435 @@
+        JsonObject jsonEventTypes = getEventTypeJson(eventType);
+
+        return serializer.toJson(jsonEventTypes);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String createEventType(Request request, Response response) {
+        endpointUtil.logReceivedRequest(logger, request);
+
+        try {
+            JsonObject jsonBody = jsonParser.parse(request.body()).getAsJsonObject();
+            JsonObject jsonEventType = jsonBody.get(JSON_EVENT_TYPE_ATTRIBUTE).getAsJsonObject();
+
+            if (jsonEventType == null) {
+                halt(HttpStatusCodes.ERROR, "no event type given in body");
+            }
+
+            EventType eventType = createEventTypeFromJson(jsonEventType);
+
+            if (eventType == null) {
+                halt(HttpStatusCodes.ERROR, "event type name already in use, or failed to parse event type");
+            } else {
+                EventPlatformFeedback feedback = EventProcessingPlatformUpdaterImpl.getInstance().registerEventType(eventType);
+
+                if (!feedback.isSuccessful()) {
+                    halt(HttpStatusCodes.ERROR, String.format("cannot register event type: %1$s", feedback.getResponseText()));
+                }
+
+                PersistenceAdapterImpl.getInstance().createArtifact(eventType, EventTypeEndpoint.getEventTypeUri(eventType.getId()));
+            }
+
+        } catch (HaltException halt) {
+            logger.info(String.format("cannot create event type: %1$s -> %2$s", halt.statusCode(), halt.body()));
+            throw halt;
+        } catch (Exception e) {
+            logger.error("cannot parse request body to event type '" + request.body() + "'");
+            logTrace(e);
+            halt(HttpStatusCodes.ERROR, e.getMessage());
+        }
+        return "";
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String deleteEventType(Request request, Response response) {
+        endpointUtil.logReceivedRequest(logger, request);
+
+        long eventTypeId = getEventTypeId(request);
+
+        EventType eventType = PersistenceAdapterImpl.getInstance().getEventType(eventTypeId);
+        String blockingEventTypes;
+        if (eventType == null) {
+            halt(HttpStatusCodes.ERROR, "cannot find event type");
+        } else {
+            if (!eventType.isDeletable()) {
+                halt(HttpStatusCodes.FORBIDDEN, "you must not delete this event type");
+            }
+
+            blockingEventTypes = getBlockingEventTypes(eventType);
+            if (blockingEventTypes.isEmpty()) {
+                boolean feedback;
+                // delete queries
+                List<EventQuery> queries = PersistenceAdapterImpl.getInstance().getEventQueries(eventTypeId);
+                EventQuery[] queryArray = new EventQuery[queries.size()];
+                feedback = PersistenceAdapterImpl.getInstance().deleteArtifacts(queries.toArray(queryArray));
+                if (!feedback) {

```

Comment:
Please remove unnecessary variable `feedback` and do an inline check in the `if`-statement